hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala

```diff
@@ -426,14 +426,14 @@ trait JSBuilderSanityChecks(instrument: Bool) extends JSBuilder:
   
   val functionParamVarargSymbol = semantics.TempSymbol(0, N, "args")
   
-  override def setupFunction(params: List[semantics.Param], body: Block)(using Raise, Scope): (Document, Document) =
+  override def setupFunction(name: Str, params: List[semantics.Param], body: Block)(using Raise, Scope): (Document, Document) =
     if instrument then
       val paramsList = params.map(p => Scope.scope.allocateName(p.sym))
       val paramsStr = Scope.scope.allocateName(functionParamVarargSymbol)
-      val checkArgsNum = doc"if ($paramsStr.length !== ${params.length}) { throw new globalThis.Error('got ' + $paramsStr.length + ' arguments, expecting ' + ${params.length}) }\n"
+      val checkArgsNum = doc"if ($paramsStr.length !== ${params.length}) { throw new globalThis.Error('Function \'$name\' expected ' + ${params.length} + ' arguments but got ' + $paramsStr.length) }\n"
```

Comment:
Please generate strings that use `"` and don't forget to escape `name` for weird characters. You can reuse the logic in `JSBuilder.makeStringLiteral` but extract it to a `JSBuilder.escapeStringCharacters` helper function.

Reply:
Thanks! I have made this and the previous changes to make function name optional.